boot/initramfs.go

```diff
@@ -54,6 +89,15 @@ func InitramfsRunModeSelectSnapsToMount(
 				blDir:  blDir,
 				blOpts: blOpts,
 			}
+			// piboot is very limited, so we need to change the kernel
```

Comment:
tryboot.txt already references a custom kernel command line that contains `kernel_status=trying` (this is via its `os_prefix=` setting). If booting fails and it reverts to using config.txt, `kernel_status` won't be present in the kernel command line. What this change in `initramfs.go` does is to find out if we were trying a kernel in the persistent configuration (`kernel_status=try`), and update it appropriately by looking at the kernel command line, the same that it is done usually by the bootloader if it has write capabilities.

I put this code here because it is run at a moment in which boot/seed partitions are already mounted but we are still in initramfs time, so we can persist kernel_status: with that we can avoid having a flag file in `/run`, and the extra complexity in GetBootVars (it would need to check for the flag, then remove it after updating kernel_status). Also, this cannot be done later because snap-bootstrap forces a reboot if it sees `kernel_status=try` (I guess it interprets that you have not really tried the kernel yet, so tries to force it).

But I agree this should be done by calling a different method from somewhere in cmd_initramfs_mounts.go? So basically, what you say in point 2, but modifying the persistent configuration (so we need to call it when boot/seed are already mounted).

Reply:
Okay, I think having this code in the initramfs is fine then, but we should move it into a different method in the boot package.

I also don't think we should necessarily be using SetBootVars with `piboot_configure: false` like this, but I would need to think on a better mechanism to inform SetBootVars that it doesn't need to modify any of the config.txt, etc stuff. But if you get rid of the EnableKernel stuff you may need to invert this 

For now I think doing it this way with SetBootVars is okay, if you just move this code into a different function inside boot rather than in InitramfsRunModeSelectSnapsToMount I think that is okay to keep moving with this branch, Samuele may have better ideas about how to organize this